File: 20242Q/poo/laboratorio/tp03/ej9/src/Rectangle.java
public class Rectangle extends Figure {

    Point topLeft;
    Point topRight;

    public Rectangle(Point tl, Point tr) {
        topLeft = tl;
        topRight = tr;
    }

    public double base() {
        return Math.abs(topRight.getX() - topLeft.getX());
    }

    public double height() {
        return Math.abs(topRight.getY() - topLeft.getY());
    }

    @Override
    public double area() {
        return base() * height();
    }

    @Override
    public double perimeter() {
        return (base() + height()) * 2;
    }

    @Override
    public String toString() {
        return "Rectángulo [ %s , %s ]".formatted(topLeft, topRight);
    }
}

File: 20242Q/poo/laboratorio/tp08/_soluciones/ej4/rectangle.rb
require_relative 'figure'

class Rectangle < Figure
  def initialize(top_left_point, bottom_right_point)
    @top_left_point = top_left_point
    @bottom_right_point = bottom_right_point
  end

  def base
    (@top_left_point.x - @bottom_right_point.x).abs
  end

  def height
    (@top_left_point.y - @bottom_right_point.y).abs
  end

  def area
    base * height
  end

  def perimeter
    (base + height) * 2
  end

  def to_s
    "Rectángulo [ #{@top_left_point} , #{@bottom_right_point} ]"
  end
end
